lib/c-api/src/wasm_c_api/module.rs

```diff
@@ -115,3 +115,364 @@ pub unsafe extern "C" fn wasm_module_serialize(
     };
     *out_ptr = byte_vec.into();
 }
+
+#[cfg(test)]
+mod tests {
+    use inline_c::assert_c;
+
+    #[test]
+    fn test_module_validate() {
+        (assert_c! {
+            #include "tests/wasmer_wasm.h"
+
+            int main() {
+                wasm_engine_t* engine = wasm_engine_new();
+                wasm_store_t* store = wasm_store_new(engine);
+
+                wasm_byte_vec_t wat;
+                wasm_byte_vec_new_from_string(&wat, "(module)");
+                wasm_byte_vec_t* wasm = wat2wasm(&wat);
+
+                assert(wasm_module_validate(store, wasm));
+
+                wasm_byte_vec_delete(wasm);
+                wasm_store_delete(store);
+                wasm_engine_delete(engine);
+
+                return 0;
+            }
+        })
+        .success();
+    }
+
+    #[test]
+    fn test_module_new() {
+        (assert_c! {
+            #include "tests/wasmer_wasm.h"
+
+            int main() {
+                wasm_engine_t* engine = wasm_engine_new();
+                wasm_store_t* store = wasm_store_new(engine);
+
+                wasm_byte_vec_t wat;
+                wasm_byte_vec_new_from_string(&wat, "(module)");
+                wasm_byte_vec_t* wasm = wat2wasm(&wat);
+
+                wasm_module_t* module = wasm_module_new(store, wasm);
+
+                assert(module);
+
+                wasm_byte_vec_delete(wasm);
+                wasm_module_delete(module);
+                wasm_store_delete(store);
+                wasm_engine_delete(engine);
+            }
+        })
+        .success();
+    }
+
+    #[test]
+    fn test_module_exports() {
+        (assert_c! {
+            #include <string.h>
+            #include "tests/wasmer_wasm.h"
+
+            void assert_exporttype_name(const wasm_exporttype_t* exporttype, const char* expected) {
+                assert(strncmp(wasm_exporttype_name(exporttype)->data, expected, strlen(expected)) == 0);
```

Comment:
This is just strcmp. strncmp + strlen isn't any safer than strcmp. Unlike rust strings where strings store the length before the string data, C strings run until the first NUL byte. Strlen will just run off the end, same as strcmp would.

Reply:
Maybe we should use `wasm_exporttype_name(exporttype)->size` instead of `strlen`. Thoughts?